app/controllers/api/databases.php

```diff
@@ -2664,14 +2664,20 @@ function updateAttribute(
 
         $database = Authorization::skip(fn() => $dbForProject->getDocument('databases', $databaseId));
 
-        if ($database->isEmpty() || (!$database->getAttribute('enabled') && $mode !== APP_MODE_ADMIN)) {
-            throw new Exception(Exception::DATABASE_NOT_FOUND);
+        $isAdminMode = $mode === APP_MODE_ADMIN;
+        $isAppUser = Auth::isAppUser(Authorization::getRoles());
+        $isPrivilegedUser = Auth::isPrivilegedUser(Authorization::getRoles());
+
+        if ($database->isEmpty() || !$database->getAttribute('enabled')) {
+            if (!($isAdminMode && ($isAppUser || $isPrivilegedUser))) {
+                throw new Exception(Exception::DATABASE_NOT_FOUND);
+            }
         }
 
         $collection = Authorization::skip(fn() => $dbForProject->getDocument('database_' . $database->getInternalId(), $collectionId));
 
         if ($collection->isEmpty() || !$collection->getAttribute('enabled')) {
-            if (!($mode === APP_MODE_ADMIN && Auth::isPrivilegedUser(Authorization::getRoles()))) {
+            if (!($isAdminMode && ($isAppUser || $isPrivilegedUser))) {
```

Comment:
I don't think admin mode is related to app user (API key) at all. If you look around here:

https://github.com/appwrite/appwrite/blob/fcd4a69ad1b3db67c6d89f02c5f230c2038a4b35/app/controllers/general.php#L317

You'll see there's never any check for $mode.